### PR TITLE
Rename setting `enable_secure_identifiers` to `enable_simple_identifiers`

### PR DESCRIPTION
**Pull request by tuanpach**
See https://github.com/ClickHouse/ClickHouse/pull/69411#discussion_r1791585855

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Setting `enable_secure_identifiers` was renamed to `enable_simple_identifiers`

### Documentation entry for user-facing changes
- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 8ad4245d732be68e14ca3895949dccf23231d865 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70613/8ad4245d732be68e14ca3895949dccf23231d865/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/70613/8ad4245d732be68e14ca3895949dccf23231d865/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: docs/en/operations/settings/settings.md [hunk 1]
```diff
@@ -2686,13 +2686,13 @@ Default value: 1
...
 
-## enable_secure_identifiers {#enable_secure_identifiers}
+## enable_simple_identifiers {#enable_simple_identifiers}
 
 Type: Bool
 
 Default value: 0
 
-If enabled, only allow secure identifiers which contain only underscore and alphanumeric characters
+If enabled, only allow simple identifiers which contain only underscore and alphanumeric characters. Usage of other characters, e.g. backslashes, quotation marks and backticks, may exploit escaping to inject SQL. This setting is recommended to enable for new installation.
```
Comment (rschu1ze) [inline docs/en/operations/settings/settings.md:2695]: `only allow simple identifier names containing alphanumeric characters and underscores only`

File: src/Core/Settings.cpp [hunk 2]
```diff
@@ -5498,8 +5498,8 @@ Replace external dictionary sources to Null on restore. Useful for testing purpo
     M(Bool, create_if_not_exists, false, R"(
 Enable `IF NOT EXISTS` for `CREATE` statement by default. If either this setting or `IF NOT EXISTS` is specified and a table with the provided name already exists, no exception will be thrown.
 )", 0) \
-    M(Bool, enable_secure_identifiers, false, R"(
-If enabled, only allow secure identifiers which contain only underscore and alphanumeric characters
+    M(Bool, enable_simple_identifiers, false, R"(
+If enabled, only allow simple identifiers which contain only underscore and alphanumeric characters
```
Comment (rschu1ze) [inline src/Core/Settings.cpp:5502]: Please keep l. 5502 in-sync with l. 2595 of settings.md

**Review by rschu1ze — APPROVED** [review]

**Comment by tuanpach** [comment]
Hi @rschu1ze 

I noticed that @vdimir  changed it to `enforce_strict_identifier_format` in [this commit](https://github.com/ClickHouse/ClickHouse/commit/bd4c3bafcb63dd0bac18dc8de29ac6851c1b7686). Should we change it to `enable_simple_identifiers`, or stick with his changes?

**Comment by rschu1ze** [comment]
@tuanpach Interesting. `strict` sounds about right for me, let's keep the naming then.

**Comment by tuanpach** [comment]
I will close this PR then.

**Pull request closed without merging**